cmd/thanos/store.go

```diff
@@ -387,6 +387,7 @@ func runStore(
 			return conf.estimatedMaxChunkSize
 		}),
 		store.WithLazyExpandedPostings(conf.lazyExpandedPostingsEnabled),
+		store.WithSeriesResort(true),
```

Comment:
I would probably go with "WithDontResort" and make resorting the default; then we dont need to pass this option here but can still expose the interface.

Reply:
fixed